src/schnorrsig.rs

```diff
@@ -142,9 +141,13 @@ impl KeyPair {
 
     /// Creates a Schnorr KeyPair directly from generic Secp256k1 secret key
     ///
-    /// Panics if internal representation of the provided [`SecretKey`] does not
-    /// holds correct secret key value obtained from Secp256k1 library
-    /// previously
+    /// # Panic
+    ///
+    /// Panics if internal representation of the provided [`SecretKey`] does not hold correct secret
+    /// key value obtained from Secp256k1 library previously, for instance when the corresponding
+    /// public key is not even key.
```

Comment:
To be clearer: this will panic on out-of-range keys (0 or in excess of the group order).